mllib/src/test/scala/org/apache/spark/ml/stat/ChiSquareTestSuite.scala

```diff
@@ -53,6 +53,20 @@ class ChiSquareTestSuite
       assert(degreesOfFreedom === Array(2, 3))
       assert(statistics ~== Vectors.dense(0.75, 1.5) relTol 1e-4)
     }
+
+    val sparseData = data.map { case LabeledPoint(label, features) =>
```

Comment:
The data doesn't have any missing 0 values; I think that's important to test here at least with some basic check. That part of the logic is tricky.

Reply:
I am OK to add a test case for this. But I am not very sure whether we should add this method for sparse dataset. Maybe the input is supposed to be dense?

Reply:
Agree. If sparse input is supported though, I think there should be a minimal test. It was supported previously right? (although there was no test?) Although it's strange, I think supporting it is OK. The most likely case where it occurs is something like: a very low dimensional vector, where the feature value 0 is common, and has been produced by a method that treats such vectors as sparse. Maybe.

Reply:
OK, I will add new testsuites for sparse input, sepecially for dataset containing column that only have zero vaues.